tiledb/api/src/query/read/output/mod.rs

```diff
@@ -697,14 +701,10 @@ impl<'data, C> Iterator for VarDataIterator<'data, C> {
         let s = self.offset_cursor;
         self.offset_cursor += 1;
 
-        if s + 1 < self.ncells {
+        if s < self.ncells {
             let start = offset_buffer[s] as usize;
             let slen = offset_buffer[s + 1] as usize - start;
             Some(&data_buffer[start..start + slen])
```

Comment:
I started poking at the examples to see if I couldn't debug the issue and stumbled over this bit. Unless my math ain't mathing right, this should just be:

```rust
let start = offsets_buffers[s] as usize;
let end = offsets_buffers[s + 1] as usize;
Some(&data_buffer[start..end]);
```

Reply:
Probably the same to the compiler but that does look simpler to the human. I'll fix it up